routetable/route_table.go

```diff
@@ -631,6 +861,64 @@ func (r *RouteTable) syncRoutesForLink(ifaceName string) error {
 	return nil
 }
 
+// deltaUpdateRoutesForLink performs deltas to the routing configuration. This does not perform a full resync, so
+// handles more routes per interface better.
+func (r *RouteTable) deltaUpdateRoutesForLink(
+	logCxt *log.Entry, linkAttrs *netlink.LinkAttrs, ifaceName string,
+	deletedCIDRsToTarget, createdCIDRsToTarget map[ip.CIDR]Target,
+) error {
+	nl, err := r.getNetlink()
+	if err != nil {
+		logCxt.Debug("Failed to connect to netlink")
+		return ConnectFailed
+	}
+
+	// Delete routes first.
+	updatesFailed := false
+	for cidr, target := range deletedCIDRsToTarget {
+		route := r.createL3Route(linkAttrs, target)
+
+		// In case this IP is being re-used, wait for any previous conntrack entry
+		// to be cleaned up.  (No-op if there are no pending deletes.)
+		r.waitForPendingConntrackDeletion(cidr.Addr())
```

Comment:
I don't think you should be waiting here.  The conntrack deletion is done async because it's slow and we don't want to block deletion itself.  Usually, IPs are reused in FIFO fashion so we take advantage of that by kicking off cleanup in the background on deletion but only waiting for it to finish when we add a new route with the same CIDR.

Also, conntrack cleanup should be done after removing a route so that no new packets can flow and create conntrack entries in parallel.